File: scanner/actions/YLEAreenaAction.js
var exec = require('child_process').exec
const Action = require('./Action')
const pythonExtensionsPath =
  __dirname + '/../../.virtualenv/bin/python ' + __dirname + '/../../python_extensions'

class YLEAreenaAction extends Action {
  process() {
    const envVars = this.envVars()
    const scriptPath =
      pythonExtensionsPath +
      '/chromecast.py --app areena --chromecast_ip ' +
      this.config.chromecast_ip +
      ' --areena_key' +
      this.config.areena_key +
      ' ' +
      this.card.uri
    exec(scriptPath, {env: envVars}, function(error, stdout, stderr) {
      console.log(stdout, stderr, error)
    })
  }
}

module.exports = YLEAreenaAction

File: scanner/actions/yleareena_action.py
from .mediaurl_action import MediaUrlAction
from utils.areena import Areena


class YLEAreenaAction(MediaUrlAction):
    def process(self):
        self.chromecast.stop()
        areena = Areena(self.config["areena_key"])
        uri = self.card["uri"]
        flag = uri.split(":")[0]
        the_rest = "".join(uri.split(":")[1:])
        # No latest or random flag = series
        if flag == "latest":
            url = areena.get_series_url_latest(the_rest)
        elif flag == "random":
            url = areena.get_series_url_random(the_rest)
            # No flag or empty flag = single program
        elif flag == "":
            url = areena.get_program_url(the_rest)
        else:
            url = areena.get_program_url(uri)
        self.card["uri"] = url
        super().process()
